app/controllers/ProjectEntryController.scala

```diff
@@ -40,6 +43,119 @@ class ProjectEntryController @Inject() (cc:ControllerComponents, config: Configu
     TableQuery[ProjectEntryRow].filter(_.id === requestedId).result.asTry
   )
 
+  protected def selectVsid(vsid: String) = dbConfig.db.run(
+    TableQuery[ProjectEntryRow].filter(_.vidispineProjectId === vsid).result.asTry
+  )
+
+  def getByVsid(vsid:String) = IsAuthenticatedAsync {uid=>{request=>
+    selectVsid(vsid).map({
+      case Success(result)=>
+        if(result.isEmpty)
+          NotFound("")
+        else
+          Ok(Json.obj("status"->"ok","result"->this.jstranslate(result.head)))
+      case Failure(error)=>
+        logger.error(error.toString)
+        InternalServerError(Json.obj("status"->"error","detail"->error.toString))
+    })
+  }}
+
+  def doUpdateTitle(requestedId:Int, newTitle:String) = selectid(requestedId).flatMap({
+      case Success(someSeq)=>
+        someSeq.headOption match {
+          case Some(record)=>
+            val updatedProjectEntry = record.copy (projectTitle = newTitle)
+            dbConfig.db.run (
+            TableQuery[ProjectEntryRow].filter (_.id === requestedId).update (updatedProjectEntry).asTry
+            )
+          case None=>
+            Future(Failure(new RecordNotFoundException(s"No record found for id $requestedId")))
+        }
+      case Failure(error)=>Future(Failure(error))
+    })
+
```

Comment:
it would be great to extract the common code here into generic methods but I can't seem to work out how.  Everything i am trying with templates seems to upset Slick which is doing its own clever-typey things behind the scenes